ge_worker/bonuses/bonus_processor.py

```diff
@@ -0,0 +1,141 @@
+import os
+import sys
+
+from datetime import datetime
+from datetime import timedelta
+from flask import g
+from logging import Logger
+
+parent_dir = os.path.abspath(os.path.join(os.getcwd(), "../app"))
+sys.path.append(parent_dir)
+sys.path.append(os.getcwd())
+
+from server import create_app, db
+from server.models.user import User
+from server.utils.credit_transfer import make_payment_transfer
+from server.utils.transfer_enums import TransferSubTypeEnum
+
+
+def get_user_by_id(user_id):
+    user = User.query.get(user_id)
+    if user:
+        return user
+    raise Exception('User with id {} not found'.format(user_id))
+
+
+def get_list_of_users_with_total_unique_outward_transactions():
+    """"
+    This method queries the database for all unique outward transactions for each user and returns a list of tuples
+    with a user id and the user's total unique outward transactions.
+    [(user_id, user_total_unique_outward_transactions)]
+    Once the list of tuples as described above is obtained, the function computes the collective unique outward
+    transactions value by iterating through the list.
+    The function then finds a user corresponding to a user id in each tuple and replaces the user id value in the
+    tuple with a user object.
+
+    returns [(user, user_total_unique_outward_transactions)]
+    """
+
+    """
+    Define SQL query with the following search criteria:
+    SEARCH CRITERIA
+        - count unique recipient user ids.
+        - credit transfer amount is greater or equal to 20wei.
+        - credit transfer occurs since 24hrs ago.
+        - credit transfer status is COMPLETE
+        - transfer subtype STANDARD
+    """
+    sql_query = '''SELECT
+    credit_transfer.sender_user_id,
+    COUNT (DISTINCT (credit_transfer.recipient_user_id))
+    FROM credit_transfer
+    LEFT JOIN transfer_account 
+    ON credit_transfer.sender_transfer_account_id = transfer_account .id
+    WHERE credit_transfer._transfer_amount_wei >= (2*10e17)
```

Comment:
Please consider replacing `2*10e7` literal with a config parameter.

Reply:
Will these config parameters be editable outside of the code ? Because the time and amount values here are eventually intended to be configurable using a GUI or with arguments provided in the cli depending on what approach we settle on.

Reply:
I think they should at a minimum be in platform config.